plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/HttpThriftMetastoreClientFactory.java

```diff
@@ -0,0 +1,176 @@
+    private final AtomicInteger chosenGetTableAlternative = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger chosenTableParamAlternative = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger chosenGetAllViewsPerDatabaseAlternative = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger chosenAlterTransactionalTableAlternative = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger chosenAlterPartitionsAlternative = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger chosenGetAllTablesAlternative = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger chosenGetAllViewsAlternative = new AtomicInteger(Integer.MAX_VALUE);
+
+    public HttpThriftMetastoreClientFactory(
+            Optional<SSLContext> sslContext,
+            Duration readTimeout,
+            ThriftHttpContext thriftHttpContext,
+            String hostname)
+    {
+        this.sslContext = requireNonNull(sslContext, "sslContext is null");
+        this.readTimeoutMillis = toIntExact(readTimeout.toMillis());
+        this.thriftHttpContext = requireNonNull(thriftHttpContext, "thriftHttpContext is null");
+        this.hostname = requireNonNull(hostname, "hostname is null");
+    }
+
+    @Inject
+    public HttpThriftMetastoreClientFactory(
+            ThriftHttpMetastoreConfig httpMetastoreConfig,
+            NodeManager nodeManager)
+    {
+        this(
+                buildSslContext(
+                        httpMetastoreConfig.isTlsEnabled(),
+                        Optional.ofNullable(httpMetastoreConfig.getKeystorePath()),
+                        Optional.ofNullable(httpMetastoreConfig.getKeystorePassword()),
+                        Optional.ofNullable(httpMetastoreConfig.getTruststorePath()),
+                        Optional.ofNullable(httpMetastoreConfig.getTruststorePassword())),
+                httpMetastoreConfig.getReadTimeout(),
+                new ThriftHttpContext(httpMetastoreConfig.getAuthenticationMode(), httpMetastoreConfig.getHttpBearerToken(), httpMetastoreConfig.getAdditionalHeaders()),
+                nodeManager.getCurrentNode().getHost());
+    }
+
+    @Override
+    public ThriftMetastoreClient create(URI uri, Optional<String> delegationToken)
+            throws TTransportException
+    {
+        return new ThriftHiveMetastoreClient(
+                () -> createHttpTransport(uri, thriftHttpContext),
+                hostname,
+                metastoreSupportsDateStatistics,
+                chosenGetTableAlternative,
+                chosenTableParamAlternative,
+                chosenGetAllTablesAlternative,
+                chosenGetAllViewsPerDatabaseAlternative,
+                chosenGetAllViewsAlternative,
+                chosenAlterTransactionalTableAlternative,
+                chosenAlterPartitionsAlternative);
```

Comment:
Maybe initialize these fields into the `ThriftHiveMetastoreClient` constructor and offer as well (for testing purposed a `ThriftHiveMetastoreClient` constructor which takes also the `chosen...` AtomicInteger parameters.
It seems unnecessary to have at the moment the constructor `ThriftHiveMetastoreClient` within this context to take this many parameters.